onnx/defs/schema.cc

```diff
@@ -447,37 +447,64 @@ void OpSchema::Finalize() {
   } while (0)
 
   // Calculate min/max number of inputs.
-  for (int i = 0; i < (int)(inputs_.size()); ++i) {
+  // <Min number of inputs> = <number of "single" inputs> + <number of "optional"
+  // but not trailing inputs>.
+  // <Max number of inputs> = <number of all inputs or std::numeric_limits<int>::max()
+  // (if the last input is variadic).
+  
+  // Flag indicates whether an optional input is trailing one (there's no single or variadic
+  // input behind).
+  bool trailingOptionalInput = true;
+  for (int i = (int)(inputs_.size() - 1); i >=0; --i) {
     switch (inputs_[i].GetOption()) {
     case OpSchema::Single:
+      trailingOptionalInput = false;
       ++min_input_;
```

Comment:
I think this code can be much more readable if you just change it to `min_input=max_input` here - basically min_input is the last non-optional input

Reply:
Great comment!!! Thank you!